store/src/java/com/zimbra/cs/service/mail/ToXML.java

```diff
@@ -2775,7 +2827,12 @@ private static Element addPart(VisitPhase phase, Element parent, Element root, M
             }
 
             try {
-                addContent(el, mpi, maxSize, neuter, defaultCharset, wantContent);
+                //checking ig content type is html
+                if (mpi.getContentType().equals("text/html") && warningMsg.length()>0 && warningMsg!=null) {
+                    addContentWithExternalWarning(el, mpi, maxSize, neuter, defaultCharset, wantContent, warningMsg);
+                }else{
+                    addContent(el, mpi, maxSize, neuter, defaultCharset, wantContent);
```

Comment:
aren't we adding warning in plain text mail?

Reply:
Plain text warning are added successfully while processing the mail at LMTP/SMTP. So we are only making changes for HTML content warning of external emails where the warning is missing(gmail domain).